src/webextension/connection.h

```diff
@@ -17,20 +17,23 @@ class Connection final : public QObject {
   Q_DISABLE_COPY_MOVE(Connection)
 
  public:
-  Connection(QObject* parent, QTcpSocket* connection);
+  Connection(QObject* parent, QIODevice* connection);
   ~Connection();
 
+  void writeMessage(QJsonObject& data);
+
+  Q_SIGNAL void onMessage(QJsonObject& message);
+
  private:
   void readData();
   void writeData(const QByteArray& data);
 
-  void writeState();
   void writeInvalidRequest();
 
   void processMessage(const QByteArray& message);
 
  private:
-  QTcpSocket* m_connection;
```

Comment:
Can't comment on the correct line but similar to my other comments the preprocessor directive names need to be updated here. e.g. `#endif // SERVERCONNECTION_H`